filebeat/scripts/tester/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+	"github.com/elastic/beats/filebeat/harvester/reader"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/match"
+)
+
+func getLogsFromFile(logfile, multiPattern string, multiNegate bool, matchMode string) ([]string, error) {
+	f, err := os.Open(logfile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	encFactory, ok := encoding.FindEncoding("utf8")
+	if !ok {
+		return nil, fmt.Errorf("unable to find 'utf8' encoding")
+	}
+
+	enc, err := encFactory(f)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize encoding: %v", err)
+	}
+
+	var r reader.Reader
+	r, err = reader.NewEncode(f, enc, 4096)
+	if err != nil {
+		return nil, err
+	}
+
+	r = reader.NewStripNewline(r)
+
+	if multiPattern != "" {
+		p := match.MustCompile(multiPattern)
```

Comment:
this panics on invalid pattern. Let's create a somewhat user-friendly error message.